pkg/api/v1alpha1/awsiamconfig_types.go

```diff
@@ -0,0 +1,125 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+// AWSIamConfig defines configuration options for AWS IAM Authenticator server
+
+// AWSIamConfigSpec defines the desired state of AWSIamConfig
+type AWSIamConfigSpec struct {
+	// AWSRegion defines a region in an AWS partition
+	AWSRegion string `json:"awsRegion,omitempty"`
+	// BackendMode defines multiple backends for aws-iam-authenticator server
+	// The server searches for mappings in order
+	BackendMode []string `json:"backendMode,omitempty"`
+	// ClusterID is a unique-per-cluster identifier for aws-iam-authenticator server
+	ClusterID string `json:"clusterID,omitempty"`
+	// MapRoles defines IAM role to a username and set of groups mapping using EKSConfigMap BackendMode
+	// Each key must match AWS EKS Style ConfigMap mapRoles
+	// +kubebuilder:validation:Optional
+	MapRoles string `json:"mapRoles,omitempty"`
+	// MapUsers defines IAM user to a username and set of groups mapping using EKSConfigMap BackendMode
+	// Each key must match AWS EKS Style ConfigMap mapUsers
+	// +kubebuilder:validation:Optional
+	MapUsers string `json:"mapUsers,omitempty"`
+	// Partition defines the AWS partition on which the IAM roles exist
+	// +kubebuilder:default:=aws
+	// +kubebuilder:validation:Optional
+	Partition string `json:"partition,omitempty"`
+}
+
+func (e *AWSIamConfigSpec) Equal(n *AWSIamConfigSpec) bool {
+	if e == n {
+		return true
+	}
+	if e == nil || n == nil {
+		return false
+	}
+	if e.AWSRegion != n.AWSRegion {
+		return false
+	}
+	if e.ClusterID != n.ClusterID {
+		return false
+	}
+	if e.MapRoles != n.MapRoles {
+		return false
+	}
+	if e.MapUsers != n.MapUsers {
+		return false
+	}
+	if e.Partition != n.Partition {
+		return false
+	}
+	return BackendModeSliceEqual(e.BackendMode, n.BackendMode)
+}
+
+func BackendModeSliceEqual(a, b []string) bool {
```

Comment:
similar question here, what's the downside to changing/adding another mode?

Reply:
Unlike the role mapping which the server watches for changes, if we were to change the backend mode we would need to restart the pods.
Again for the initial version I have kept them mutable.